staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_standard.go

```diff
@@ -141,7 +141,11 @@ func (as *availabilitySet) DetachDisk(diskName, diskURI string, nodeName types.N
 			(disk.ManagedDisk != nil && diskURI != "" && strings.EqualFold(*disk.ManagedDisk.ID, diskURI)) {
 			// found the disk
 			klog.V(2).Infof("azureDisk - detach disk: name %q uri %q", diskName, diskURI)
-			disks[i].ToBeDetached = to.BoolPtr(true)
+			if as.cloud.Environment.Name == "AZURESTACKCLOUD" {
+				disks = append(disks[:i], disks[i+1:]...)
+			} else {
+				disks[i].ToBeDetached = to.BoolPtr(true)
```

Comment:
also add an unit test to catch this change?

Reply:
fixed.

Reply:
I tried, that would require a complicated mock, @Sakuralbj could you help address this in another PR? thanks.